RecoHI/HiCentralityAlgos/plugins/HiHFFilterProducer.cc

```diff
@@ -0,0 +1,103 @@
+
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "RecoLocalCalo/CaloTowersCreator/src/CaloTowerCandidateCreator.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerDetId.h"
+#include "DataFormats/CaloTowers/interface/CaloTower.h"
+#include "DataFormats/CaloTowers/interface/CaloTowerCollection.h"
+#include "DataFormats/HeavyIonEvent/interface/HFFilterStruct.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescriptionFiller.h"
+
+#include "DataFormats/Common/interface/Wrapper.h"
+
+#include <cmath>
+
+class HiHFFilterProducer : public edm::stream::EDProducer<> {
+public:
+  explicit HiHFFilterProducer(const edm::ParameterSet&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  edm::EDGetTokenT<CaloTowerCollection> srcTowers_;
+};
+
+HiHFFilterProducer::HiHFFilterProducer(const edm::ParameterSet& iConfig)
+    : srcTowers_(consumes<CaloTowerCollection>(iConfig.getParameter<edm::InputTag>("srcTowers"))) {
+  produces<pat::HFFilterStruct>("hiHFfilters");
+}
+
+void HiHFFilterProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace std;
+  using namespace edm;
+
+  auto const& towers = iEvent.getHandle(srcTowers_);
```

Comment:
`auto const& towers = event.get(srcTowers_);` should be fine, or do you specifically want a Handle instead of the underlying object?

Reply:
implemented in the commit f5397ec4d90e6b54c5f6b056650ebb6cb334036c